lib/navigation/preprocess_encoders.cpp

```diff
@@ -33,10 +49,86 @@ core::EncoderData EncodersPreprocessor::detectOutliers(const core::EncoderData e
   -also has to be able to handle 1 unreliable sensor
   -also figure out return type/ what we update and update
   documentation as appropriate
+  }
   */
-  return {0, 0, 0, 0};
+template<typename T>
```

Comment:
not quite:
```
template<std::size_t N>
std::array<core::Float,3> EncodersPreprocessor::quartiles(std::array<std::uint32_t , N> & encoder_data){ ...
```

Reply:
This is I think a better use of templates and then you can make the code adaptable to it taking an array of length 3 or 4 (in code knumiums or knumimus-1 because no hard coding please)